slither/printers/summary/evm.py

```diff
@@ -0,0 +1,136 @@
+                    node_source_line = contract_file[0:node.source_mapping['start']].count("\n".encode("utf-8")) + 1
+                    print('\t\tSource line {}: {}'.format(node_source_line, contract_file_lines[node_source_line-1].rstrip()))
+                    print('\t\tEVM Instructions:')
+                    node_pcs = contract_pcs.get(node_source_line, [])
+                    for pc in node_pcs:
+                        print('\t\t\t0x{:x}: {}'.format(int(pc), contract_cfg.get_instruction_at(pc)))
+                        
+            for modifier in contract.modifiers:
+                print(f'\tModifier {modifier.canonical_name}')
+                for node in modifier.nodes:
+                    node_source_line = contract_file[0:node.source_mapping['start']].count("\n".encode("utf-8")) + 1
+                    print('\t\tSource line {}: {}'.format(node_source_line, contract_file_lines[node_source_line-1].rstrip()))
+                    print('\t\tEVM Instructions:')
+                    node_pcs = contract_pcs.get(node_source_line, [])
+                    for pc in node_pcs:
+                        print('\t\t\t0x{:x}: {}'.format(int(pc), contract_cfg.get_instruction_at(pc)))
+
+                        
+    def _extract_evm_info(self, slither):
+        """
+        Extract evm information for all derived contracts using evm_cfg_builder
+
+        Returns: evm CFG and Solidity source to Program Counter (pc) mapping
+        """
+        
+        evm_info = {}
+        
+        for contract in slither.contracts_derived:
+            contract_bytecode_runtime = slither.crytic_compile.bytecode_runtime(contract.name)
+            contract_srcmap_runtime = slither.crytic_compile.srcmap_runtime(contract.name)
+            cfg = CFG(contract_bytecode_runtime)
+            evm_info['cfg', contract.name] = cfg
+            evm_info['mapping', contract.name] = self._generate_source_to_evm_ins_mapping(cfg.instructions,
+                                                                                 contract_srcmap_runtime, slither,
+                                                                                 contract.source_mapping['filename_absolute'])
+            contract_bytecode_init= slither.crytic_compile.bytecode_init(contract.name)
+            contract_srcmap_init = slither.crytic_compile.srcmap_init(contract.name)
+            cfg_init = CFG(contract_bytecode_init)
+            evm_info['cfg_init', contract.name] = cfg_init
+            evm_info['mapping_init', contract.name] = self._generate_source_to_evm_ins_mapping(cfg_init.instructions,
+                                                                                contract_srcmap_init, slither,
+                                                                                contract.source_mapping['filename_absolute'])
+        return(evm_info)
+
+    
+    def _generate_source_to_evm_ins_mapping(self, evm_instructions, srcmap_runtime, slither, filename):
```

Comment:
Consider moving this function to another module, maybe `slither/evm/convert.py`?

Reply:
I have made all the changes suggested above. Thanks.